src/mono/mono.proj

```diff
@@ -615,14 +648,14 @@
   <!-- General targets -->
   <Target Name="BuildMono" AfterTargets="Build" DependsOnTargets="$(MonoDependsOnTargets)">
     <PropertyGroup Condition="'$(BuildMonoAOTCrossCompilerOnly)' != 'true'">
-      <_MonoRuntimeFilePath Condition="'$(TargetsWindows)' == 'true'">$(MonoObjDir)out\bin\monosgen-2.0.dll</_MonoRuntimeFilePath>
-      <_MonoRuntimeFilePath Condition="'$(TargetsOSX)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.dylib</_MonoRuntimeFilePath>
-      <_MonoRuntimeFilePath Condition="'$(TargetsMacCatalyst)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.dylib</_MonoRuntimeFilePath>
-      <_MonoRuntimeFilePath Condition="'$(TargetsiOS)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.dylib</_MonoRuntimeFilePath>
-      <_MonoRuntimeFilePath Condition="'$(TargetstvOS)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.dylib</_MonoRuntimeFilePath>
-      <_MonoRuntimeFilePath Condition="'$(TargetsBrowser)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.a</_MonoRuntimeFilePath>
-      <_MonoRuntimeFilePath Condition="'$(_MonoRuntimeFilePath)' == ''">$(MonoObjDir)out\lib\libmonosgen-2.0.so</_MonoRuntimeFilePath>
-      <_MonoRuntimeStaticFilePath Condition="'$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true' or '$(TargetsAndroid)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.a</_MonoRuntimeStaticFilePath>
+      <_MonoRuntimeFilePath Condition="'$(TargetsWindows)' == 'true'">$(MonoObjDir)out\bin\$(MonoSharedLibName).dll</_MonoRuntimeFilePath>
+      <_MonoRuntimeFilePath Condition="'$(TargetsOSX)' == 'true'">$(MonoObjDir)out\lib\lib$(MonoSharedLibName).dylib</_MonoRuntimeFilePath>
+      <_MonoRuntimeFilePath Condition="'$(TargetsMacCatalyst)' == 'true'">$(MonoObjDir)out\lib\lib$(MonoSharedLibName).dylib</_MonoRuntimeFilePath>
+      <_MonoRuntimeFilePath Condition="'$(TargetsiOS)' == 'true'">$(MonoObjDir)out\lib\lib$(MonoSharedLibName).dylib</_MonoRuntimeFilePath>
+      <_MonoRuntimeFilePath Condition="'$(TargetstvOS)' == 'true'">$(MonoObjDir)out\lib\lib$(MonoSharedLibName).dylib</_MonoRuntimeFilePath>
+      <_MonoRuntimeFilePath Condition="'$(TargetsBrowser)' == 'true'">$(MonoObjDir)out\lib\lib$(MonoLibName).a</_MonoRuntimeFilePath>
```

Comment:
```suggestion
      <_MonoRuntimeFilePath Condition="'$(TargetsBrowser)' == 'true'">$(MonoObjDir)out\lib\lib$(MonoSharedLibName).a</_MonoRuntimeFilePath>
```

Reply:
Should be MonoLibName since since this is the static archive and it can be named different compared to the shared library on some platforms where we still use coreclr.so|dylib|dll.

Reply:
We need to build shared library using final name directly since we will link towards that library when building the component shared libraries, if not we need to run platform specific tooling changing library name in imports etc.

Reply:
Hopefully we can drop the use of coreclr.so|dylib|dll at some point and then MonoLibName == MonoShareLibName, but until then they can be different.

Reply:
I'm not sure I agree with @thaystg: mono is not a shared library on wasm.

I see that above `MonoSharedLibName` is set to `MonoLibName` even for `$(TargetsBrowser)` (and I find that a little surprising) so it's the same value.

But I think it's better to use SharedLibName when talking about the shared library only.

Reply:
Okay, but why we cannot use two lines like this?
```
<_MonoRuntimeFilePath Condition="'$(TargetsBrowser)' == 'false'>$(MonoObjDir)out\lib\$(MonoLibName).a</_MonoRuntimeFilePath>
<_MonoRuntimeFilePath Condition="'$(TargetsBrowser)' == 'true'>$(MonoObjDir)out\lib\$(LibPrefix)$(MonoSharedLibName)$(LibExt)</_MonoRuntimeFilePath>
```

Reply:
Since I did some other changes related to review feedback I reworked this as well.